### PR TITLE
test(e2e): add cy commands for storage-browser

### PR DESCRIPTION
**Pull request by ashwinkumar6**
#### Description of changes
- Follow up PR to [address comment](https://github.com/aws-amplify/amplify-ui/pull/6147#discussion_r1852947313) to refactor common cy utils to commands.
- The `upload a folder` test is also cleaned up to only performed required checks
- Add TS doc string to `cypress/support/commands.d.ts`
- update fileuploader e2e tests to use cy.fileInputUpload

#### Description of how you validated changes
Tested cypress locally

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ashwinkumar6** [comment]
previous comments on [this PR](https://github.com/aws-amplify/amplify-ui/pull/6202)

File: packages/e2e/features/ui/components/storage/storage-browser/action-menu.feature [hunk 1]
```diff
@@ -118,12 +118,10 @@ Feature: Create folder with Storage Browser
     # upload file
     Then I see the "Upload" menuitem
     When I click the "Upload" menuitem
-    Then the "Upload" button is disabled
     Then I upload a folder "e2eTemp" with "2" files with random names
     Then I see "Not started"
     Then I click the label containing text "Overwrite existing files"
     When I click the "Upload" button
-    Then I see "100%"
```
Comment (ashika112) [inline packages/e2e/features/ui/components/storage/storage-browser/action-menu.feature:126]: why is this removed?

File: packages/e2e/features/ui/components/storage/storage-browser/action-menu.feature [hunk 2]
```diff
@@ -135,7 +133,4 @@ Feature: Create folder with Storage Browser
     Then I click the "Delete" menuitem
     Then I click the "Delete" button
     Then I see "All files deleted"
-    When I click the "Exit" button
```
Comment (ashika112) [inline packages/e2e/features/ui/components/storage/storage-browser/action-menu.feature:138]: ditto?

File: packages/e2e/features/ui/components/storage/storage-browser/action-menu.feature (hunk 1 above)
Comment (ashika112) [reply packages/e2e/features/ui/components/storage/storage-browser/action-menu.feature:126]: Nvm. it is covered in another test. Thanks

**Review by ashika112 — APPROVED** [review]

**Review by jordanvn — APPROVED** [review]
